backup/README.md

```diff
@@ -0,0 +1,157 @@
+
+## Deploy Velero
+
+You can run the installation script `terraform/velero-install.sh` as-is, or you can run the individual commands yourself if you want to see more details on what the installation involves. First the script will install Velero components to the UpCloud Kubernetes Service cluster, and then it adds the necessary CSI plugin to Velero. Lastly, the volumesnapshotclass needs to be labeled with a Velero tag to make Velero aware of the CSI snapshot feature.
+
+Kick off the installation script:
+
+```text
+cd terraform
+./velero-install.sh
+```
+
+Velero should be installed in a few seconds. You can check the installation logs in the Velero pod. You should see a successful connection to the S3 bucket:
+
+```text
+kubectl logs -n velero velero-pod-name
+```
+
+Now we are ready to deploy a test app on the cluster!
+
+## Deploy a Test App
+
+Our test app is a simple nginx pod. We use a Persistent Volume for the nginx logs, so we can follow website loads and see if our backup and restore works. View the app yaml for more information. Velero is also capable of doing pre- and post-hooks for both backups and restores. The test app has an example on how to run scripts before and after a backup is run.
+
+```text
+cat k8s/velero-demo-app.yaml
+```
+
+Deploy the app:
+
+```text
+kubectl apply -f k8s/velero-demo-app.yaml
+```
+
+Once the load balancer service is up, load the default nginx welcome page:
+
+```text
+kubectl get services -n velero-demo
+curl -i lb-yourlbdnsname-1.upcloudlb.com
+```
+
+Verify that you can see the webpage loads in nginx logs. The logs are stored on a Persistent Volume:
+
+```text
+kubectl exec -n velero-demo nginx-pod-name -it -- cat /var/log/nginx/access.log
+```
+
+Now we are ready to create a backup! Note the time stamps on the nginx logs, so you can verify the backup and restore worked as expected. We are using a label to select the test app deployment and the PVC, but Velero can also backup the whole velero-demo namespace or even the whole cluster.
+
+```text
+velero backup create velerotest1 --selector app=velero-app 
+```
+
+Let's reload the nginx welcome page a few times to get more data. Remember, we created the backup point-in-time before these log messages, so they should not be visible after restore.
+
+```text
+curl -i lb-yourlbdnsname-1.upcloudlb.com
+kubectl exec -n velero-demo nginx-pod-name -it -- cat /var/log/nginx/access.log
+```
+
+Since we ran pre- and post-hooks, there should be two text files, `prehook.txt` and `posthook.txt`, in the log folder as well:
+
+```text
+kubectl exec -n velero-demo nginx-pod-name -it -- ls -la /var/log/nginx/
+```
+
+We have data in the logs, and our backup has completed with pre- and post-hooks executed. Let's delete the test app deployment and the PVC and see if we can do a successful restore:
+
+```text
+kubectl delete deployments.apps -n velero-demo nginx
+kubectl delete pvc -n velero-demo nginx-logs
```

Comment:
This leaves the related PV still behind, as the reclaim policy is "Retain" by default. I think this is why my first restore had a warning:

```
time="2023-06-29T06:55:55Z" level=warning msg="Cluster resource restore warning: could not restore, PersistentVolume \"pvc-791221a7-1bcd-433f-bc20-139ec2c8f7d7\" already exists. Warning: the in-cluster version is different than the backed-up version." logSource="pkg/controller/restore_controller.go:505" restore=velero/velerotest1-20230629095554
```

Should we delete the PV as well for demo purposes?

Reply:
Fixed now, Reclaim policy changed to Delete and the PV is deleted in the demo (and restored)